### PR TITLE
Fixed pandas SettingWithCopy warning

### PR DESCRIPTION
**Pull request by jaclynbeck-sage**
This update gets rid of the multiple `SettingWithCopy` warnings generated by `calculate_distribution` in transform.py. 

I think the issue was that setting `df = df[df[is_scored] == "Y"]` just prior to the code that generated the warnings created a "slice" of the dataframe, and pandas behavior is unpredictable when setting things in that case. My fix gets rid of the warning by explicitly making `df` a real data frame instead of a slice, with `df.copy()`. 

I also fixed another warning in this function, about `.append()` being deprecated on pandas series. I used `pd.concat()` as suggested by the warning instead.

Lastly, there was a really unreadable piece of code in this function that I'm assuming got split by a python code formatter or something:
```
    obj["distribution"][
        0
    ] -= 1  # since this was calculated with the artificial 0 value, we subtract it
    obj["distribution"][
        -1
    ] -= 1  # since this was calculated with the artificial upper_bound, we subtract it
```
I put things back on one line. 

I've confirmed that the distribution JSON file output after my fixes is identical to the one from before my fixes.

### CODE REVIEW

**Comment by jaclynbeck-sage** [comment]
@JessterB hold off on review/approval until I merge `dev` back into this branch so it meshes with all of Brad's changes!

**Comment by jaclynbeck-sage** [comment]
@JessterB it should be ready for review now. `dev` has been merged in and there don't seem to be any issues or conflicts.

File: src/agoradatatools/etl/transform.py
Comment (thomasyu888) [inline src/agoradatatools/etl/transform.py:120]: Nit: This is great - I know there are two other places that use `.append` in `create_proteomics_distribution_data`.  Should we switch those to`pd.concat` in this PR, or create a Jira ticket for that?

**Review by thomasyu888 — APPROVED** [review]
🚀 Thanks for doing these!  Going to pre-approve since my only comment is "Nit"

File: src/agoradatatools/etl/transform.py
Comment (JessterB) [reply src/agoradatatools/etl/transform.py:120]: Whichever @jaclynbeck-sage prefers works for me.

Comment (jaclynbeck-sage) [reply src/agoradatatools/etl/transform.py:120]: The `create_proteomics_distribution_data` is using `.append` on a list, not a pandas dataframe, so these appends are ok to leave in!

Comment (thomasyu888) [reply src/agoradatatools/etl/transform.py:120]: Thanks for clarifying!  Should have looked closer!

**Pull request merged**
